server/schedulers/balance_leader.go

```diff
@@ -105,55 +96,49 @@ func (l *balanceLeaderScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
 func (l *balanceLeaderScheduler) Schedule(cluster opt.Cluster) []*operator.Operator {
 	schedulerCounter.WithLabelValues(l.GetName(), "schedule").Inc()
 
-	stores := cluster.GetStores()
-
-	// source/target is the store with highest/lowest leader score in the list that
-	// can be selected as balance source/target.
-	source := l.selector.SelectSource(cluster, stores)
-	target := l.selector.SelectTarget(cluster, stores)
+	leaderScheduleStrategy := l.opController.GetLeaderScheduleStrategy()
+	sources := filter.SelectSourceStores(cluster.GetStores(), l.filters, cluster)
```

Comment:
Maybe you should implement `filterStoresBy` in a safe way. It is hard for people to understand why we should not get stores first.

Reply:
em, how about rename the function to `InplaceSelectSourceStores`?

Reply:
I prefer a stupid but safe way, however, it is OK for me if you insist.